### PR TITLE
Drop any modal

### PR DESCRIPTION
**Pull request by jivey**

### CODE REVIEW

**Comment by jivey** [comment]
I have a  bunch of type check errors to sort out, but want to push this up to get help debugging a circular JSON issue with the API.

File: tutor/src/screens/assignment-review/drop-question-modal.tsx [hunk 1]
```diff
@@ -13,58 +18,142 @@ const DetailsLink = styled(TutorLink)`
...
+                        <Icon type="external-link-square" />
+                    </DetailsLink>
+                    <p>
+                        Drop question for all students in all sections. Question will be marked
+                        as dropped but remain available to students.
+                    </p>
+                    <Box margin={{ bottom: 'large' }}>
+                        <b>Reallocate Points</b>
+                        <Box direction="column" margin={{ left: '40px' }}>
+                            <Box margin={{ bottom: 'large' }}>
```
Comment (nathanstitt) [inline tutor/src/screens/assignment-review/drop-question-modal.tsx:82]: This should be `as="label"` as well right?  same as it's sibling below?

File: tutor/src/screens/assignment-review/drop-question-modal.tsx [hunk 2]
```diff
@@ -13,58 +18,142 @@ const DetailsLink = styled(TutorLink)`
...
+                                <Box pad="top" margin="right">
+                                    <RadioInput standalone name="type"
+                                        id="full-credit" value="full"
+                                        checked={Boolean(drop && drop.drop_method == 'full_credit')}
+                                        onChange={({ target: { checked } }: { target: { checked: boolean } }) => {
+                                            checked && (drop.setDropMethod('full_credit'))
+                                        }} />
+                                </Box>
+                                <Box as="label" htmlFor="full-credit" direction="column" margin={{ left: '8px' }}>
```
Comment (nathanstitt) [inline tutor/src/screens/assignment-review/drop-question-modal.tsx:91]: the parent is a `label`, does this need to be?

File: tutor/src/screens/assignment-review/drop-question-modal.tsx [hunk 3]
```diff
@@ -13,58 +18,142 @@ const DetailsLink = styled(TutorLink)`
...
+                            <Box as="label">
+                                <Box pad="top" margin="right">
+                                    <RadioInput standalone name="type"
+                                        id="zero-credit" value="zero"
+                                        checked={Boolean(drop && drop.drop_method == 'zeroed')}
+                                        onChange={({ target: { checked } }: { target: { checked: boolean } }) => {
+                                            checked && (drop.setDropMethod('zeroed'))
+                                        }} />
+                                </Box>
+                                <Box as="label" htmlFor="zero-credit" direction="column" margin={{ left: '8px' }}>
```
Comment (nathanstitt) [inline tutor/src/screens/assignment-review/drop-question-modal.tsx:105]: same as above, probably doesn't need to be a label

**Review by nathanstitt — APPROVED** [review]
looks great Josiah

**Pull request merged**